src/components/cc-pricing-product-consumption/cc-pricing-product-consumption.js

```diff
@@ -268,142 +328,124 @@ export class CcPricingProductConsumption extends withResizeObserver(LitElement)
         };
       });
 
-      this._simulator = new PricingConsumptionSimulator(this.sections);
+      this._simulator = new PricingConsumptionSimulator(this.product.sectionList);
     }
   }
 
   render () {
+    return html`
+      ${this.product.state === 'error' ? html`
+        <cc-notice intent="danger" message=${i18n('cc-pricing-product-consumption.error')}></cc-notice>
+      ` : ''}
 
-    const someNullishIntervals = this.sections
-      ?.some(({ intervals }) => intervals == null) ?? true;
-    const everyQuantityAtZero = this.sections
-      ?.map(({ type }) => this._simulator.getQuantity(type))
-      ?.every((quantity) => quantity === 0) ?? true;
+      ${this.product.state === 'loading' ? html`
+        <cc-loader></cc-loader>
+      ` : ''}
 
-    const name = this.name ?? SKELETON_NAME;
+      ${this.product.state === 'loaded' ? this._renderLoaded() : ''}
+    `;
+  }
 
+  _renderLoaded () {
     const bodyClasses = {
       'body--big': (this._size > 600),
       'body--small': (this._size <= 600),
     };
 
-    return html`
-      <slot name="head">
-        <div class="head">
-          <div class="head-info">
-            <cc-img class="product-logo" src="${ifDefined(this.icon ?? undefined)}" ?skeleton=${this.icon == null}></cc-img>
-            <div class="name-wrapper">
-              <span class="name ${classMap({ skeleton: (this.name == null) })}">${name}</span>
-            </div>
-          </div>
-
-          ${!this.error ? html`
-            <slot></slot>
-          ` : ''}
-        </div>
-        <hr>
-      </slot>
+    const someNullishIntervals = this.product.sectionList
+      ?.some(({ intervals }) => intervals == null) ?? true;
+    const everyQuantityAtZero = this.product.sectionList
+      ?.map(({ type }) => this._simulator.getQuantity(type))
+      ?.every((quantity) => quantity === 0) ?? true;
 
+    return html`
       <div class="body ${classMap(bodyClasses)}">
-
-        ${this.error && (this.sections == null) ? html`
-          <cc-error class="error-global">${i18n('cc-pricing-product-consumption.error')}</cc-error>
-          <hr>
-        ` : ''}
-
-        ${this.sections?.map((section) => html`
-          ${this._renderSection(section)}
-          <hr>
-        `)}
-
+        ${this.product.sectionList?.map((section) => this._renderSection(section))}
```

Comment:
suggestion: if the subrender function always gets a `sectionList` typed as `PricingSection[]`, there's no need to use `?.`